core/server/master/src/main/java/alluxio/master/file/meta/ReadOnlyInode.java

```diff
@@ -0,0 +1,212 @@
+  }
+
+  @Override
+  public long getId() {
+    return mDelegate.getId();
+  }
+
+  @Override
+  public long getTtl() {
+    return mDelegate.getTtl();
+  }
+
+  @Override
+  public TtlAction getTtlAction() {
+    return mDelegate.getTtlAction();
+  }
+
+  @Override
+  public long getLastModificationTimeMs() {
+    return mDelegate.getLastModificationTimeMs();
+  }
+
+  @Override
+  public String getName() {
+    return mDelegate.getName();
+  }
+
+  @Override
+  public short getMode() {
+    return mDelegate.getMode();
+  }
+
+  @Override
+  public PersistenceState getPersistenceState() {
+    return mDelegate.getPersistenceState();
+  }
+
+  @Override
+  public long getParentId() {
+    return mDelegate.getParentId();
+  }
+
+  @Override
+  public String getOwner() {
+    return mDelegate.getOwner();
+  }
+
+  @Override
+  public boolean isDeleted() {
+    return mDelegate.isDeleted();
+  }
+
+  @Override
+  public boolean isDirectory() {
+    return mDelegate.isDirectory();
+  }
+
+  @Override
+  public boolean isFile() {
+    return mDelegate.isFile();
+  }
+
+  @Override
+  public boolean isPinned() {
+    return mDelegate.isPinned();
+  }
+
+  @Override
+  public boolean isPersisted() {
+    return mDelegate.isPersisted();
+  }
+
+  @Override
+  public String getUfsFingerprint() {
+    return mDelegate.getUfsFingerprint();
+  }
+
+  @Override
+  public AccessControlList getACL() {
+    return mDelegate.getACL();
+  }
+
+  @Override
+  public DefaultAccessControlList getDefaultACL() throws UnsupportedOperationException {
+    return mDelegate.getDefaultACL();
+  }
+
+  @Override
+  public FileInfo generateClientFileInfo(String path) {
+    return mDelegate.generateClientFileInfo(path);
+  }
+
+  @Override
+  public boolean checkPermission(String user, List<String> groups, AclAction action) {
+    return mDelegate.checkPermission(user, groups, action);
+  }
+
+  @Override
+  public AclActions getPermission(String user, List<String> groups) {
+    return mDelegate.getPermission(user, groups);
+  }
+
+  @Override
+  public InodeMeta.Inode toProto() {
+    return mDelegate.toProto();
+  }
+
+  @Override
+  public JournalEntry toJournalEntry() {
+    return mDelegate.toJournalEntry();
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (!(other instanceof ReadOnlyInode)) {
+      return false;
+    }
+    ReadOnlyInode otherInode = (ReadOnlyInode) other;
+    return mDelegate.equals(otherInode.mDelegate);
+  }
+
+  @Override
+  public int hashCode() {
+    return mDelegate.hashCode();
+  }
+
+  /**
+   * Casts the inode as an {@link ReadOnlyInodeDirectory} if it is one, otherwise throws an
+   * exception.
+   *
+   * This gives convenience in method chaining, e.g.
+   *
+   * inode.asDirectory().getChildren()
+   *
+   * instead of
+   *
+   * ((ReadOnlyInodeDirectory) inode).getChildren()
+   *
+   * @return the inode as an inode directory
+   */
+  public ReadOnlyInodeDirectory asDirectory() {
```

Comment:
Should we make an `asFile` method? Also, should we always use `as` instead of casting (we still cast in a few places)

Reply:
done